### PR TITLE
:seedling: Update dependencies for 2021/06/07

### PR DESCRIPTION
**Pull request by randomvariable**
Updating dependencies as a follow up to https://github.com/kubernetes-sigs/cluster-api/pull/4746

### CODE REVIEW

File: hack/tools/go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module sigs.k8s.io/controller-runtime/hack/tools
 go 1.15
 
 require (
-	github.com/golangci/golangci-lint v1.23.6
-	github.com/joelanford/go-apidiff v0.0.0-20191206194835-106bcff5f060
-	sigs.k8s.io/controller-tools v0.4.0
+	github.com/golangci/golangci-lint v1.40.1
+	github.com/joelanford/go-apidiff v0.1.0
+	sigs.k8s.io/controller-tools v0.5.0
```
Comment (vincepri) [inline hack/tools/go.mod:8]:
```suggestion
	sigs.k8s.io/controller-tools v0.6.0-beta.0
```

File: go.mod [hunk 2]
```diff
@@ -3,19 +3,19 @@ module sigs.k8s.io/controller-runtime
...
 	github.com/go-logr/logr v0.4.0
 	github.com/go-logr/zapr v0.4.0
 	github.com/googleapis/gnostic v0.5.5 // indirect
 	github.com/hashicorp/golang-lru v0.5.4 // indirect
 	github.com/imdario/mergo v0.3.12 // indirect
-	github.com/onsi/ginkgo v1.16.2
-	github.com/onsi/gomega v1.12.0
-	github.com/prometheus/client_golang v1.10.0
+	github.com/onsi/ginkgo v1.16.4
+	github.com/onsi/gomega v1.13.0
```
Comment (vincepri) [inline go.mod:14]: Do we need the new minor for gomega?

File: go.mod [hunk 3]
```diff
@@ -3,19 +3,19 @@ module sigs.k8s.io/controller-runtime
...
+	github.com/prometheus/client_golang v1.11.0
 	github.com/prometheus/client_model v0.2.0
 	go.uber.org/goleak v1.1.10
-	go.uber.org/zap v1.16.0
+	go.uber.org/zap v1.17.0
```
Comment (vincepri) [inline go.mod:18]: Same as above, is this needed? I'd rather not change it so close to release

File: go.mod [hunk 4]
```diff
@@ -3,19 +3,19 @@ module sigs.k8s.io/controller-runtime
 go 1.16
 
 require (
-	github.com/evanphx/json-patch v4.9.0+incompatible
+	github.com/evanphx/json-patch v4.11.0+incompatible
```
Comment (vincepri) [inline go.mod:6]: Checked the release notes, this should be good to bump

File: go.mod (hunk 3 above)
Comment (randomvariable) [reply go.mod:18]: There's two updates for CVEs transitively with x/crypto and yaml

File: go.mod (hunk 2 above)
Comment (randomvariable) [reply go.mod:14]: Nothing exciting here. CAPA is using some of the deprecated pieces, but doesn't look like CAPI is.

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Pull request merged**